src/reporter.ts

```diff
@@ -0,0 +1,180 @@
+/*
+ *  Copyright 2021 EPAM Systems
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+import RPClient from '@reportportal/client-javascript';
+import { Reporter, TestCase, TestResult } from '@playwright/test/reporter';
+import { Attribute, ReportPortalConfig } from './models';
+import { TEST_ITEM_TYPES } from './constants';
+import { getConfig } from './utils';
+import { StartLaunchObjType, StartTestObjType, FinishTestItemObjType } from './models/reporting';
+
+const promiseErrorHandler = (promise: Promise<any>, message = '') =>
+  promise.catch((err) => {
+    console.error(message, err);
+  });
+
+export interface TestItem {
+  id: string;
+  name: string;
+  status?: string;
+  attributes?: Attribute[];
+  description?: string;
+}
+
+interface Suite {
+  id: string;
+  name: string;
+  path?: string;
+  attributes?: Attribute[];
+}
+
+interface TestResp extends TestCase {
+  parent: {
+    title: string;
+    _isDescribe: boolean;
```

Comment:
Does it a good way to use private key?

Reply:
I not found any other way to implement case suite in suite.